docs/the-new-architecture/pillars-fabric-components.md

```diff
@@ -518,6 +516,223 @@ Differently from Native Components, Fabric requires us to manually implement the
 
 #### Android
 
+Android follows some similar steps to iOS. We have to generate the code for Android, and then we have to write some native code to make it works.
+
+##### Generate the Code - Android
+
+To generate the code for Android, we need to manually invoke the CodeGen. This is done similarly to what we did for iOS: first, we need to add the package to the app and then we need to invoke a script.
```

Comment:
That's something we were discussing even some time back. It is actually impossible to write a TM or a FC without invoking the CodeGen somehow. That's because the dev has to use some types and includes that are generated by the CodeGen. 
So, we either assume that they have full knowledge of what is going to be produced (something that, to me, is a little bit too strong, especially if they are approaching the development of a library for the first time) or we give them a way to manually generate the code and we tell them what to look for. I prefer this approach, because: 

> Give a man a fish, he will eat for one day. Teach them how to fish, and they will eat for the rest of their life.

And they will probably come back with fewer issues if they learn where to find the code generated by the codegen.

What do you think?

Reply:
> And, when the app will build, it will fail building because the component is not implementing the code it requires.

I'm not sure I follow here, but I think this is not the case.

What I mean is that is totally reasonable to:

1. Setup the Codegen Config for a specific module
1. Write the .js/.ts spec file in that module
1. Run `react-native run-android`.
1. Build will be green as you're running the App but not dependening on the generate code. Codegen will run though as it was configured and a spec file was found.
1. You then code around the generated code.
1. When done with the extra component, you'll run `react-native run-android` once more.

This integrates really well with the IDE as well, as we're not letting the build fail so Android Studio/IntelliJ will properly discover the generated code after the first `run-android` and will allow to build around it.

Reply:
I'm not sure it will work. The structure we are working with is the following:
```
root
'-> App
'-> FabricComponent
```
Following your steps, we will have
1. Setup the `FabricComponent/android/build.gradle` file with the codegen
2. Write the .js/.ts spec file in that module in the `FabricComponent/js` folder
3. Run `react-native run-android`

but we have to run this command from the App folder and, at the moment, the app does not know anything about the `FabricComponent` because we have not add to the app yet. Or am I missing something?